pkg/sriov/list_test.go

```diff
@@ -0,0 +1,349 @@
+	testCases := []struct {
+		testNetwork   []*NetworkBuilder
+		nsName        string
+		listOptions   []v1.ListOptions
+		expectedError error
+		client        bool
+	}{
+		{
+			testNetwork:   []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			nsName:        defaultNetNsName,
+			expectedError: nil,
+			client:        true,
+		},
+		{
+			testNetwork:   []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			nsName:        "",
+			expectedError: fmt.Errorf("failed to list sriov networks, 'nsname' parameter is empty"),
+			client:        true,
+		},
+		{
+			testNetwork: []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			nsName:      defaultNetNsName,
+			listOptions: []v1.ListOptions{{LabelSelector: "test"}},
+			client:      true,
+		},
+		{
+			testNetwork:   []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			nsName:        defaultNetNsName,
+			listOptions:   []v1.ListOptions{{LabelSelector: "test"}, {Continue: "true"}},
+			expectedError: fmt.Errorf("error: more than one ListOptions was passed"),
+			client:        true,
+		},
+		{
+			testNetwork:   []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			nsName:        defaultNetNsName,
+			expectedError: fmt.Errorf("failed to list sriov networks, 'apiClient' parameter is empty"),
+			client:        false,
+		},
+	}
+	for _, testCase := range testCases {
+		var testSettings *clients.Settings
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(buildDummySrIovNetworkObject())
+		}
+
+		netBuilders, err := List(testSettings, testCase.nsName, testCase.listOptions...)
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil && len(testCase.listOptions) == 0 {
+			assert.Equal(t, len(netBuilders), len(testCase.testNetwork))
+		}
+	}
+}
+
+func TestNetworkCleanAllNetworksByTargetNamespace(t *testing.T) {
+	testCases := []struct {
+		testNetwork    []*NetworkBuilder
+		operatorNsName string
+		targetNsName   string
+		listOptions    []v1.ListOptions
+		client         bool
+		expectedError  error
+	}{
+		{
+			testNetwork:    []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			operatorNsName: defaultNetNsName,
+			targetNsName:   defaultNetTargetNsName,
+			client:         true,
+		},
+		{
+			testNetwork:   []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			targetNsName:  defaultNetTargetNsName,
+			expectedError: fmt.Errorf("failed to clean up sriov networks, 'operatornsname' parameter is empty"),
+			client:        true,
+		},
+		{
+			testNetwork:    []*NetworkBuilder{buildValidSriovNetworkTestBuilder(buildTestClientWithDummyObject())},
+			operatorNsName: defaultNetNsName,
+			expectedError:  fmt.Errorf("failed to clean up sriov networks, 'targetnsname' parameter is empty"),
```

Comment:
Can you explicitly add targetNsName: "" so that it is clear why you are expecting the error that you are?